src/NUnitEngine/nunit.engine/Extensibility/ExtensionSelector.cs

```diff
@@ -0,0 +1,75 @@
+﻿// ***********************************************************************
```

Comment:
Functionality in this new class mostly extracted from ExtensionAssembly.cs - some changes to consider targetframework. Refactor was for test purposes.